src/libos/src/net/socket/unix/stream/file.rs

```diff
@@ -83,10 +90,17 @@ impl File for Stream {
         Ok(())
     }
 
-    fn poll(&self) -> Result<PollEventFlags> {
-        warn!("poll is not supported for unix_socket");
-        let events = PollEventFlags::empty();
-        Ok(events)
+    fn poll_new(&self) -> IoEvents {
+        match &*self.inner() {
+            // linux return value
+            Status::Idle(info) => IoEvents::OUT | IoEvents::HUP,
+            Status::Connected(endpoint) => endpoint.poll(),
+            Status::Listening(_) => IoEvents::empty(),
```

Comment:
It seems to me that the poll for a listening socket is not implemented. This is a very common functionality. I think we can add it later. But a big warning should be printed here to cry out loudly that this feature is missing.

Reply:
It returns nothing when you poll a listener.